serverlessworkflow/modules/ROOT/pages/service-orchestration/orchestration-of-openapi-based-services.adoc

```diff
@@ -0,0 +1,423 @@
+            }
+          }
+        },
+        {
+          "name": "multiply",
+          "functionRef": {
+            "refName": "multiplication", <4> 
+            "arguments": { <5> 
+               "leftElement": ".difference",
+               "rightElement": ".multiplyValue"
+            }
+          }
+        }
+      ],
+      "end": {
+        "terminate": "true"
+      }
+    }
+  ]
+}
+----
+
+<1> The Operation State definition
+<2> Reference to the `subtraction` function
+<3> Definition of the `subtraction` function arguments
+<4> Reference to the `multiplication` function
+<5> Definition of the `multiplication` function arguments
+
+Any link:{spec_doc_url}#Action-Definition[workflow actions] can call a function definition, having a similar way of referencing the functions that you previously defined in the workflow.
+
+To map the function's arguments, refer to the parameters described in the OpenAPI specification file, link:{open_api_spec_url}#operation-object[operations section]. In this example, the parameters are part of the request body.
+
+The Kogito engine will map the parameter naming according to the OpenAPI specification. Then, you can use {understanding-jq-expressions}[JQ expressions] to extract the payload data and map it to the required parameters.
+
+See the PetStore API example:
+
+.Petstore OpenAPI specification excerpt
+[source,json]
+----
+{
+  "/pet/{petId}": {
+      "get": {
+        "tags": [
+          "pet"
+        ],
+        "summary": "Find pet by ID",
+        "description": "Returns a single pet",
+        "operationId": "getPetById",
+        "parameters": [
+          {
+            "name": "petId",
+            "in": "path",
+            "description": "ID of pet to return",
+            "required": true,
+            "schema": {
+              "type": "integer",
+              "format": "int64"
+            }
+          }
+        ]
+     }
+  }
+}
+----
+
+[NOTE]
+====
+Only the highlighted excerpt of the OpenAPI specification file is shown in this example.
+====
+
+In this example, only one parameter named `petId` goes in the path. In this case, this is what an invocation to this function looks like:
+
+.Example of calling the PetStore function
+[source,json]
+----
+{
+  "name": "CallPetStore", <1>
+  "actionMode": "sequential",
+  "type": "operation",
+  "actions": [
+    {
+      "name": "getPet",
+      "functionRef": {
+        "refName": "getPetById", <2> 
+        "arguments": { <3>
+          "petId": ".petId"
+        }
+      }
+    }
+  ]
+}
+----
+
+[NOTE]
+====
+Only the highlighted excerpt of the workflow definition file is shown in this example.
+====
+
+<1> The state definition
+<2> The function definition reference, assuming that you have a function definition named `getPetById` for the PetStore OpenAPI specification
+<3> The arguments definition. In this example, Kogito will add the argument `petId` to the request path before making the call.
+
+== Running the Example
```

Comment:
Wondering if explanations of how to run a specific example should be more part of the example readme (as this seems specific to the example)?

Reply:
It's part of the example readme, but since we reference this example through the guide, why not a quick procedure showing how to run it? I don't have a strong opinion about this. Could maybe others chime in? @ibek wdyt?

Reply:
I'll remove it. It will be a pain to keep it up to date with the example README. people can go there and follow the instructions to run the example.